flow360/component/project.py

```diff
@@ -0,0 +1,560 @@
+            project._root_webapi = RestApi(VolumeMeshInterfaceV2.endpoint, id=root_asset.id)
+        return project
+
+    def _check_initialized(self):
+        """
+        Checks if the project instance has been initialized correctly.
+
+        Raises
+        ------
+        Flow360ValueError
+            If the project is not initialized.
+        """
+        if not self.metadata or not self.solver_version:
+            raise Flow360ValueError(
+                "Project is not initialized - use Project.from_file or Project.from_cloud"
+            )
+
+    def get_root_simulation_json(self) -> dict:
+        """
+        Returns the default simulation JSON for the project based on the root asset type.
+
+        Returns
+        -------
+        dict
+            The simulation JSON for the project.
+
+        Raises
+        ------
+        Flow360ValueError
+            If the root item type or ID is missing from project metadata.
+        Flow360WebError
+            If the simulation JSON cannot be retrieved.
+        """
+        self._check_initialized()
+        root_type = self.metadata.root_item_type
+        root_id = self.metadata.root_item_id
+        if not root_type or not root_id:
+            raise Flow360ValueError("Root item type or ID is missing from project metadata")
+        resp = self._root_webapi.get(method="simulation/file", params={"type": "simulation"})
+        if not isinstance(resp, dict) or "simulationJson" not in resp:
+            raise Flow360WebError("Couldn't retrieve default simulation JSON for the project")
+        simulation_json = json.loads(resp["simulationJson"])
+        return simulation_json
+
+    # pylint: disable=too-many-arguments, too-many-locals
+    def _run(
+        self,
+        params: SimulationParams,
+        target: AssetOrResource,
+        draft_name: str = None,
+        fork: bool = False,
+        run_async: bool = True,
+    ):
+        """
+        Runs a simulation for the project.
+
+        Parameters
+        ----------
+        params : SimulationParams
+            The simulation parameters to use for the run
+        target : AssetOrResource
+            The target asset or resource to run the simulation against.
+        draft_name : str, optional
+            The name of the draft to create for the simulation run (default is None).
+        fork : bool, optional
+            Indicates if the simulation should fork the existing case (default is False).
+        run_async : bool, optional
+            Specifies whether the simulation should run asynchronously (default is True).
+
+        Returns
+        -------
+        AssetOrResource
+            The destination asset
+
+        Raises
+        ------
+        Flow360ValueError
+            If the simulation parameters lack required length unit information, or if the
+            root asset (Geometry or VolumeMesh) is not initialized.
+        """
+        defaults = self.get_root_simulation_json()
```

Comment:
Can we somehow cache the root item's JSON in project? If a user call _run() in a big loop we will need a lot of repeated API calls.

Reply:
Fixed in [076877a](https://github.com/flexcompute/Flow360/pull/511/commits/076877aa3adaa7aefc56b1e85c7f5f38fa1a789f)